src/core/Directus/Services/AssetService.php

```diff
@@ -77,35 +82,14 @@ public function getAsset($fileHashId, array $params = [])
         }
         $file = $result->current()->toArray();
 
-        $ext = pathinfo($file['filename'], PATHINFO_EXTENSION);
-        $uploadedFileName = $file['id'].'.'.$ext;
-
-        if(count($params) == 0) {
-           $url=get_file_root_url();
-           $img = $this->filesystem->read($uploadedFileName);
-           $result=[];
-           $result['mimeType']=Image::make($this->filesystem->read($uploadedFileName))->mime();
```

Comment:
I added some logic in here that would retrieve the real last modified date from disk: https://github.com/directus/api/blob/76709d5ebb63ac839256f20f68cdb5c9487882de/src/core/Directus/Services/AssetService.php#L88-L92